hvcc/generators/ir2c/SignalExpr.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (C) 2022-2025 Daniel Billotte, Wasted Audio
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Dict, List
+
+from hvcc.types.IR import IRSignalList
+
+from .expr.expr_c_writer import ExprCWriter
+from .HeavyObject import HeavyObject
+
+
+class SignalExpr(HeavyObject):
+    """Handles the math objects.
+    """
+
+    preamble = "cExprSig"
+
+    obj_eval_functions: Dict = {}
```

Comment:
This turns out to be problematic. When you load hvcc as a library and try to access it multiple times this Dict is being reused and ends up accumulating outdated objects.

This became apparent when I tried to run more than two tests using `expr~`

Reply:
We can reset the value before running `ir2c.compile()` but it feels dirty and a bit of a bad design.

Reply:
Went with it anyway.